specification/sbi_atee.adoc

```diff
@@ -13,6 +43,858 @@ running a VMM).
+|===
+
+=== Function: TEE TVM Add Shared Memory Region (FID #19)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_add_shared_memory_region(unsigned long tvm_guest_id,
+                                                   unsigned long tvm_gpa_addr,
+                                                   unsigned long region_len);
+-----
+Marks the range of TVM physical address space starting at `tvm_gpa_addr` as reserved
+for the mapping of shared memory. The region is initially unpopulated, and pages of
+shared memory may be inserted by calling `sbi_tee_tvm_add_shared_pages()`. Attempts
+by a TVM vCPU to access an unpopulated region will cause a `shared_page_fault` exit
+from `sbi_tee_tvm_run_vcpu()`. Both `tvm_gpa_addr` and `region_len` must be 4kB-aligned.
+The `tvm_guest_id is the unique identifier returned by a previous call to`sbi_tee_create_tvm()`.
+Shared memory regions may only be added to TVMs prior to the call to `sbi_tee_tvm_finalize()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_add_shared_memory_region_errors]
+.TEE TVM Add Shared Memory Region
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully
+| SBI_ERR_INVALID_PARAM | The input parameters were invalid, or the TVM wasn't
+                          in the correct state
+|===
+
+=== Function: TEE TVM Add Shared Pages (FID #20)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_add_shared_pages(unsigned long tvm_guest_id,
+                                           unsigned long base_address,
+                                           unsigned long page_type,
+                                           unsigned long num_pages,
+                                           unsigned long tvm_guest_gpa);
+
+-----
+Maps num_pages of non-confidential memory at `base_address` to non-confidential
+memory at `tvm_guest_gpa`. The mapping must lie within a region previously created with
+`sbi_tee_tvm_add_shared_memory_region()`. Shared pages can be mapped-in on demand at any
+point of time. The `tvm_guest_id is the unique identifier returned by a previous call to
+`sbi_tee_create_tvm()`.
+
+The permissible values for the `page_type` parameter appear below.
+[source, C]
+-------------------
+enum tsm_page_type {
+    /* 4KiB */
+    PAGE_4K = 0,
+    /* 2 MiB */
+    PAGE_2MB = 1,
+    /* 1 GiB */
+    PAGE_1GB = 2,
+    /* 512 GiB */
+    PAGE_512GB = 3,
+}
+-------------------
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_add_shared_pages_errors]
+.TEE TVM Add Shared Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully
+| SBI_ERR_INVALID_PARAM | One or more parameters were invalid
+|===
+
+== TEE AIA Extension (EID #0x54414941)
+
```

Comment:
Probably should include some prose here about what this extension is for?

Also, this is still very WIP :)

Reply:
@abrestic-rivos I had added the following blurb...do we need more?
```
The TEE Interrupt extension supplements the TEE Host extension with hardware-assisted interrupt
virtualization using the RISC-V Advanced Interrupt Architecture (AIA) on platforms which
support it.
```